alf/algorithms/rnd_algorithm.py

```diff
@@ -91,29 +124,38 @@ def train_step(self,
             TrainStep:
                 outputs: empty tuple ()
                 state: empty tuple ()
-                info: RNDInfo
+                info: ICMInfo
         """
         observation = time_step.observation
+
+        if self._stacked_frames:
+            # Assuming stacking in the last dim, we only keep the last frame.
+            observation = observation[..., -1:]
+
         if self._observation_normalizer is not None:
             observation = self._observation_normalizer.normalize(observation)
 
+        if self._encoder_net is not None:
+            observation = tf.stop_gradient(self._encoder_net(observation)[0])
+
         pred_embedding, _ = self._predictor_net(observation)
         target_embedding, _ = self._target_net(observation)
 
-        loss = 0.5 * tf.reduce_mean(
+        loss = tf.reduce_sum(
```

Comment:
why change to reduce_sum?

Reply:
That's what I usually have for experiments

Reply:
Is this related to why a small learning rate can be used for training with good results in your experiments?

Reply:
I never used a small learning rate? This is not learning improvements, just RND